pkg/object/infos.go

```diff
@@ -6,19 +6,38 @@ package object
 import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/cli-runtime/pkg/resource"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
 )
 
 func InfoToUnstructured(info *resource.Info) *unstructured.Unstructured {
 	return info.Object.(*unstructured.Unstructured)
 }
 
-func UnstructuredToInfo(obj *unstructured.Unstructured) *resource.Info {
+func UnstructuredToInfo(obj *unstructured.Unstructured) (*resource.Info, error) {
+	annos, found, err := unstructured.NestedStringMap(obj.Object, "metadata", "annotations")
+	if err != nil {
+		return nil, err
+	}
+
+	source := "unstructured"
+	if found {
+		path, ok := annos[kioutil.PathAnnotation]
+		if ok {
+			source = path
+			delete(annos, kioutil.PathAnnotation)
+			err := unstructured.SetNestedStringMap(obj.Object, annos, "metadata", "annotations")
```

Comment:
Too low level here: better to use `unstructure.SetAnnotations()`.

Reply:
actually, there is: `obj.Object.SetAnnotations(annos)`.

Reply:
https://github.com/kubernetes/kubernetes/blob/323f34858de18b862d43c40b2cced65ad8e24052/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go#L412